cdcv2/metadata/model.go

```diff
@@ -14,129 +14,217 @@
-}
-
-// -------------------- About owner schedule -------------------- //
-// 1. ControllerObservation.SetOwner puts an owner on a given capture;
-// 2. ControllerObservation.SetOwner can also stop an owner;
-// 3. Capture fetches owner launch/stop events with CaptureObservation.OwnerChanges;
-// 4. Capture calls Capture.PostOwnerRemoved when the owner exits;
-// 5. After controller confirms the old owner exits, it can re-reschedule it.
-// -------------------------------------------------------------- //
-
-// ---------- About changefeed processor captures schedule ---------- //
-// 1. ControllerObservation.SetProcessors attaches some captures to a changefeed;
-// 2. ControllerObservation.SetProcessors can also detach captures from a changefeed;
-// 3. Owner calls OwnerObservation.ProcessorChanges to know processors are created;
-// 4. Capture fetches processor launch/stop events with CaptureObservation.ProcessorChanges;
-// 5. How to rolling-update a changefeed with only one worker capture:
-//    * controller needs only to attach more captures to the changefeed;
-//    * it's owner's responsibility to evict tables between captures.
-// 5. What if owner knows processors are created before captures?
-//    * table schedule should be robust enough.
-// ------------------------------------------------------------------ //
-
-// ---------------- About keep-alive and heartbeat ---------------- //
-// 1. Capture updates heartbeats to metadata by calling CaptureObservation.Heartbeat,
-//    with a given timeout, for example, 1s;
-// 2. On a capture, controller, owners and processors share one same Context, which is
-//    associated with deadline 10s. CaptureObservation.Heartbeat will refresh the deadline.
-// 3. Controller is binded with a lease (10+1+1)s, for deadline, heartbeat time-elapsed
-//    and network clock skew.
-// 4. Controller needs to consider re-schedule owners and processors from a capture,
-//    if the capture has been partitioned with metadata storage more than lease+5s;
-// ---------------------------------------------------------------- //
+	ChangefeedIdent
+
+	UpstreamID uint64 `gorm:"column:upstream_id;type:bigint(20) unsigned;not null;index:upstream_id,priority:1" json:"upstream_id"`
+	SinkURI    string `gorm:"column:sink_uri;type:text;not null" json:"sink_uri"`
+	StartTs    uint64 `gorm:"column:start_ts;type:bigint(20) unsigned;not null" json:"start_ts"`
+	TargetTs   uint64 `gorm:"column:target_ts;type:bigint(20) unsigned;not null" json:"target_ts"`
+	// Note that pointer type is used here for compatibility with the old model, and config should never be nil in practice.
+	Config *config.ReplicaConfig `gorm:"column:config;type:longtext;not null" json:"config"`
+}
+
+type ChangefeedState struct {
+	ChangefeedUUID ChangefeedUUID      `gorm:"column:changefeed_uuid;type:bigint(20) unsigned;primaryKey" json:"changefeed_uuid"`
+	State          model.FeedState     `gorm:"column:state;type:text;not null" json:"state"`
```

Comment:
the type should be `enum`

Reply:
We will handle these changes in another PR.